### PR TITLE
Test p2p routing using kademlia DHT.

### PR DESCRIPTION
**Pull request by arijitAD**
## Changes

- Test p2p routing using kademlia DHT.

## Tests

```
TestKadDHT()
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [ ] All integration tests and required coverage checks are passing

## Issues

- #520

### CODE REVIEW

File: dot/network/host.go [hunk 1]
```diff
@@ -76,8 +76,16 @@ func newHost(ctx context.Context, cfg *Config) (*host, error) {
 		return nil, err
 	}
 
+	dhtOpts := []kaddht.Option{
+		kaddht.Datastore(dsync.MutexWrap(ds.NewMapDatastore())),
+		kaddht.Mode(kaddht.ModeAutoServer),
```
Comment (noot) [inline dot/network/host.go:81]: curious why this is changed?

File: dot/network/host_test.go [hunk 2]
```diff
@@ -509,3 +513,78 @@ func TestStreamCloseMetadataCleanup(t *testing.T) {
 	_, ok = info.handshakeData[nodeB.host.id()]
 	require.False(t, ok)
 }
+
+func createServiceHelper(t *testing.T, num int) []*Service {
+	t.Helper()
```
Comment (noot) [inline dot/network/host_test.go:518]: oh this is cool!

**Review by noot — APPROVED** [review]
looks good to me!

File: dot/network/host.go (hunk 1 above)
Comment (arijitAD) [reply dot/network/host.go:81]: Default DHT mode is `ModeAuto` which starts it as a client and it cannot respond to queries. https://github.com/libp2p/go-libp2p-kad-dht/blob/09d923fcf68218181b5cd329bf5199e767bd33c3/dht_options.go#L28

**Review by RyRy79261 — APPROVED** [review]
Looks great!

**Pull request merged**